frontend/src/metabase/visualizations/components/settings/ChartSettingFieldPicker.jsx

```diff
@@ -69,52 +73,109 @@ const ChartSettingFieldPicker = ({
+      bg="bg-white"
+      align="center"
     >
-      {showDragHandle && <GrabberHandle name="grabber" noPointer noMargin />}
-      {showColorPicker && seriesKey && (
-        <FieldPickerColorPicker
-          pillSize="small"
-          value={colors[seriesKey]}
-          onChange={value => {
-            onChangeSeriesColor(seriesKey, value);
-          }}
-        />
-      )}
       <ChartSettingSelect
-        value={value}
+        data-testid="chartsettings-field-picker-select"
+        pl="xs"
+        pr="xs"
+        w="100%"
+        isInitiallyOpen={autoOpenWhenUnset && value === undefined}
         options={options}
+        value={value}
         onChange={onChange}
-        placeholder={t`Select a field`}
+        icon={
+          showDragHandle || (showColorPicker && seriesKey) ? (
+            <Group noWrap spacing="xs" pl="sm" pr="xs">
+              {showDragHandle && (
+                <GrabberHandle
+                  name="grabber"
+                  noMargin
+                  onClick={e => e.stopPropagation()}
+                  c="text-medium"
+                  className={CS.pointerEventsAll}
+                />
+              )}
+              {showColorPicker && seriesKey && (
+                <ChartSettingColorPicker
+                  pillSize="small"
+                  value={colors[seriesKey]}
+                  onChange={value => {
+                    onChangeSeriesColor(seriesKey, value);
+                  }}
+                  className={CS.pointerEventsAll}
+                />
+              )}
+            </Group>
+          ) : null
+        }
         placeholderNoOptions={t`No valid fields`}
-        isInitiallyOpen={autoOpenWhenUnset && value === undefined}
-        hiddenIcons
+        placeholder={t`Select a field`}
+        iconWidth="auto"
+        rightSectionWidth="auto"
+        rightSection={
+          <Group noWrap spacing="sm" p="xs" mr="sm">
+            {!disabled && (
+              <ActionIcon c="text-medium" size="sm" radius="xl" p={0}>
+                <Icon name="chevrondown" />
+              </ActionIcon>
+            )}
+            {menuWidgetInfo && (
+              <ChartSettingActionIcon
+                icon="ellipsis"
+                data-testid={`settings-${value}`}
+                onClick={e => {
+                  onShowWidget(menuWidgetInfo, e);
+                }}
+              />
+            )}
+            {onRemove && (
+              <ChartSettingActionIcon
+                icon="close"
+                data-testid={`remove-${value}`}
+                onClick={onRemove}
+              />
+            )}
+          </Group>
+        }
+        styles={{
+          wrapper: {
+            display: "flex",
+          },
+          icon: {
+            position: "static",
+            width: "auto",
+          },
+          input: {
+            "&[data-with-icon]": {
+              paddingLeft: 0,
+            },
+            marginLeft: theme.spacing.xs,
```

Comment:
note: I think Mantine 6 supports using CSS variables for `--mantine-spacing-xs` here too, if you want to avoid using `useTheme` here: https://v6.mantine.dev/theming/mantine-provider/#css-variables

<img src="https://github.com/user-attachments/assets/4f203d48-4d09-41c1-9642-2bbd83db0c06" width="400">